packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -429,69 +501,74 @@ export const ColorScaleBlock: FC<BlockProps> = ({ appBridge }) => {
-                                        index={index}
-                                        width={currentlyDraggedColorId === color.id ? 0 : width}
-                                        height={colorScaleHeight}
-                                        className={joinClassNames([
-                                            isFirst ? COLOR_SQUARE_FIRST_ELEMENT_CLASSES : '',
-                                            isLast ? COLOR_SQUARE_LAST_ELEMENT_CLASSES : '',
-                                        ])}
-                                        color={color}
-                                        onDrop={handleDrop}
-                                        onDelete={handleColorSquareDelete}
-                                        onResizeStart={handleResizeStart}
-                                        canDragAndDrop={!resizedColorIndex.current}
-                                        isEditing={isEditing}
-                                        setCurrentlyDraggedColorId={setCurrentlyDraggedColorId}
-                                        currentlyDraggedColorId={currentlyDraggedColorId}
-                                    />
-                                </div>
-                            );
-                        })}
-
-                        {displayableItems.length === 0 && (
-                            <EmptyView
-                                height={
-                                    blockSettings.customHeight ? blockSettings.heightInput : blockSettings.heightSlider
-                                }
-                            />
-                        )}
-                    </DndProvider>
-                </div>
+                {displayableItems.length > 0 && (
+                    <div
+                        ref={colorScaleBlockInnerRef}
+                        style={{
+                            height: colorScaleHeight,
+                        }}
+                        className="tw-rounded tw-w-full tw-flex"
+                        // Note: onMouseUp and handleResize are defined here intentionally, instead of being in the DragHandle component.
+                        // The reason for this is that the dragging feature stops working if I move these to DragHandle,
+                        // perhaps because the component is being destroyed on every re-render and causing issues with dragging.
+                        // The 'handleResizeStart' method, on the other hand, needs to stay in DragHandle, because it needs to
+                        // identify which color square is being resized.
+                    >
+                        <DndProvider backend={HTML5Backend}>
+                            {displayableItems.map((color: ColorProps, index: number) => {
+                                const width = color.width
+                                    ? color.width
+                                    : calculateDefaultColorWidth(displayableItems.length, colorScaleBlockRef);
+
```

Comment:
```suggestion
                    const width = color.width ?? calculateDefaultColorWidth(displayableItems.length, colorScaleBlockRef);

```

Reply:
you could even add here the id check. this would lead to
`const width = currentlyDraggedColorId === color.id ? 0 : color.width ?? calculateDefaultColorWidth(displayableItems.length, colorScaleBlockRef)`

Reply:
Excellent suggestion!